src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,70 +1,107 @@
-import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import { ColorLegend } from "../../visualizations/chart/legend/ColorLegend";
-import PushpinSizeLegend from "./legends/PushpinSizeLegend";
-import PushpinCategoryLegend from "../geoChart/legends/PushpinCategoryLegend";
 
-import { useNumbericSymbols } from "../base/hook/TranslationsHook";
 import { generateLegendColorData } from "./geoChartColor";
+import PushpinCategoryLegend from "./legends/PushpinCategoryLegend";
+import PushpinSizeLegend from "./legends/PushpinSizeLegend";
+import { useNumbericSymbols } from "../base/hook/TranslationsHook";
+import { ColorLegend } from "../../visualizations/chart/legend/ColorLegend";
+import { BOTTOM } from "../../visualizations/chart/legend/PositionTypes";
+import { IGeoData, IPushpinCategoryLegendItem } from "../../../interfaces/GeoChart";
+import { getAvailableLegends } from "../../../helpers/geoChart/data";
 
 export interface IGeoChartLegendRendererProps {
-    config: IGeoConfig;
-    geoData: IGeoData;
+    categoryItems?: IPushpinCategoryLegendItem[]; // used for Category legend
+    format?: string;
+    geoData?: IGeoData; // used for Color/Size legend
+    height?: number;
+    locale?: string;
     position?: string;
-    categoryItems?: IPushpinCategoryLegendItem[];
-    onItemClick?(item: IPushpinCategoryLegendItem): void;
+    responsive?: boolean;
+    showFluidLegend?: boolean;
+    onItemClick?: (item: IPushpinCategoryLegendItem) => void;
 }
 
 export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    const numericSymbols = useNumbericSymbols();
-    const { position = TOP, categoryItems, geoData } = props;
-    const { size, color } = geoData;
-
-    const hasSizeData = Boolean(size && size.data.length);
-    const hasSegmentData = Boolean(categoryItems && categoryItems.length);
-    const isColorLegendVisible = Boolean(color && color.data.length && !hasSegmentData);
-    const isLegendVisible = isColorLegendVisible || hasSizeData || hasSegmentData;
+    const { categoryItems, geoData, position, responsive, showFluidLegend } = props;
+    const { hasCategoryLegend, hasColorLegend, hasSizeLegend } = getAvailableLegends(categoryItems, geoData);
 
+    const isLegendVisible = hasCategoryLegend || hasColorLegend || hasSizeLegend;
     if (!isLegendVisible) {
         return null;
     }
 
-    const classes = cx("geo-legend s-geo-legend", `position-${position}`, { "has-size-legend": hasSizeData });
+    const isFluidLegendEnabled = responsive && showFluidLegend;
+    const isBottomPosition = isFluidLegendEnabled || position === BOTTOM;
+
+    const classNames = cx("geo-legend", "s-geo-legend", {
+        "viz-fluid-legend-wrap": isFluidLegendEnabled,
+        "viz-static-legend-wrap": !isFluidLegendEnabled,
+        "has-size-legend": hasSizeLegend,
+        [`position-${position}`]: !isBottomPosition,
+        "position-bottom": isBottomPosition,
```

Comment:
I think [`position-${position}`]: true that is enough, please check it

Reply:
this is required in case: 
- position is different than bottom
- isFluidLegendEnabled is true 

https://github.com/gooddata/gooddata-react-components/pull/1466/files/e7d433000c9214e253833c668b47de17b9066fda#diff-de6ab458c862f711882696296630b813R37

Reply:
I have added a comment for this case in code